metricflow-semantics/metricflow_semantics/model/semantics/metric_lookup.py

```diff
@@ -31,11 +34,63 @@
 class MetricLookup:
     """Tracks semantic information for metrics by linking them to semantic models."""
 
+    @staticmethod
+    def create(  # noqa: D102
+        semantic_manifest: SemanticManifest,
+        semantic_model_lookup: SemanticModelLookup,
+        custom_granularities: Dict[str, ExpandedTimeGranularity],
+    ) -> MetricLookup:
+        manifest_object_lookup = SemanticManifestObjectLookup(semantic_manifest)
+
+        linkable_spec_index_builder = LinkableSpecIndexBuilder(
+            semantic_manifest=semantic_manifest,
+            semantic_model_lookup=semantic_model_lookup,
+            manifest_object_lookup=manifest_object_lookup,
+            max_entity_links=MAX_JOIN_HOPS,
+        )
+        linkable_spec_index = linkable_spec_index_builder.build_index()
+        linkable_spec_resolver = ValidLinkableSpecResolver(
```

Comment:
Should we just call `MetricLookup.create_using_index()` at this point to avoid the duplicate code?

Reply:
Updated.